circle.yml

```diff
@@ -0,0 +1,33 @@
+machine:
+  ruby:
+    version: '2.1.0'
+
+dependencies:
+  cache_directories:
+    - 'vendor/bundle_32'
+    - 'vendor/bundle_40'
+    - 'vendor/bundle_41'
+    - 'vendor/bundle_42'
+  override:
+    - bundle check --path=vendor/bundle_32 --gemfile Gemfile.rails32 || bundle install --jobs=4 --retry=3 --gemfile Gemfile.rails32 --path=vendor/bundle_32
+    - bundle check --path=vendor/bundle_40 --gemfile Gemfile.rails40 || bundle install --jobs=4 --retry=3 --gemfile Gemfile.rails40 --path=vendor/bundle_40
+    - bundle check --path=vendor/bundle_41 --gemfile Gemfile.rails41 || bundle install --jobs=4 --retry=3 --gemfile Gemfile.rails41 --path=vendor/bundle_41
+    - bundle check --path=vendor/bundle_42 --gemfile Gemfile.rails42 || bundle install --jobs=4 --retry=3 --gemfile Gemfile.rails42 --path=vendor/bundle_42
+
+test:
+  override:
+    - bundle check --path=vendor/bundle_32 && bundle exec rake test:
```

Comment:
Because all these tests run on the same host, I have to do the `check` here so I can set the `--path`, which Bundler then [remembers](http://bundler.io/v1.3/man/bundle-install.1.html#REMEMBERED-OPTIONS) for subsequent commands.